src/FsLex.Core/fslexast.fs

```diff
@@ -6,8 +6,8 @@ open System.Collections.Generic
 open System.Globalization
 open FSharp.Text.Lexing
 
-type Ident = string
-type Code = string * Position
+type Ident = string * Range
```

Comment:
the primary change is to add a Range to the various identifier/text structures, and from there flow the effects of that change up to things like error reporting.